src/ProjectTemplates/BlazorTemplates.Tests/BlazorServerTemplateTest.cs

```diff
@@ -47,21 +57,32 @@ public async Task BlazorServerTemplateWorks_NoAuth()
             var buildResult = await Project.RunDotNetBuildAsync();
             Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", Project, buildResult));
 
+            var browser = Fixture.BrowserManager.IsAvailable(browserKind) ?
+                await Fixture.BrowserManager.GetBrowserInstance(browserKind, BrowserContextInfo) :
+                null;
+
             using (var aspNetProcess = Project.StartBuiltProjectAsync())
             {
                 Assert.False(
                     aspNetProcess.Process.HasExited,
                     ErrorMessages.GetFailedProcessMessageOrEmpty("Run built project", Project, aspNetProcess.Process));
 
                 await aspNetProcess.AssertStatusCode("/", HttpStatusCode.OK, "text/html");
-                if (BrowserFixture.IsHostAutomationSupported())
+
+                if (Fixture.BrowserManager.IsAvailable(browserKind))
                 {
-                    aspNetProcess.VisitInBrowser(Browser);
-                    TestBasicNavigation();
+                    var page = await aspNetProcess.VisitInBrowserAsync(browser);
+                    await TestBasicNavigation(page);
+                    await page.CloseAsync();
                 }
                 else
                 {
-                    BrowserFixture.EnforceSupportedConfigurations();
+                    Assert.False(
```

Comment:
Can we commonize this pattern? `EnsureBrowserAvailable(BrowserKind)`?

Reply:
done